sdks/fabric/interoperation-node-sdk/src/AssetManager.ts

```diff
@@ -0,0 +1,523 @@
+    assetExchangeAgreement.setId(assetID);
+    assetExchangeAgreement.setRecipient(recipientECert);
+    assetExchangeAgreement.setLocker(lockerECert);
+    return assetExchangeAgreement.serializeBinary().toString();
+}
+
+// Create a fungible asset exchange agreement structure
+function createFungibleAssetExchangeAgreementSerialized(assetType, numUnits, recipientECert, lockerECert)
+{
+    const assetExchangeAgreement = new assetLocksPb.FungibleAssetExchangeAgreement();
+    assetExchangeAgreement.setType(assetType);
+    assetExchangeAgreement.setNumunits(numUnits);
+    assetExchangeAgreement.setRecipient(recipientECert);
+    assetExchangeAgreement.setLocker(lockerECert);
+    return assetExchangeAgreement.serializeBinary().toString();
+}
+
+// Create an asset lock structure
+function createAssetLockInfoSerialized(hashValue, expiryTimeSecs)
+{
+    const lockInfoHTLC = new assetLocksPb.AssetLockHTLC();
+    lockInfoHTLC.setHashbase64(hashValue);
+    lockInfoHTLC.setExpirytimesecs(expiryTimeSecs);
+    lockInfoHTLC.setTimespec(assetLocksPb.AssetLockHTLC.TimeSpec.EPOCH)
+    const lockInfoHTLCSerialized = lockInfoHTLC.serializeBinary();
+    const lockInfo = new assetLocksPb.AssetLock();
+    lockInfo.setLockmechanism(assetLocksPb.LockMechanism.HTLC);
+    lockInfo.setLockinfo(lockInfoHTLCSerialized);
+    return lockInfo.serializeBinary().toString();
+}
+
+// Create an asset claim structure
+function createAssetClaimInfoSerialized(hashPreimage)
+{
+    const claimInfoHTLC = new assetLocksPb.AssetClaimHTLC();
+    claimInfoHTLC.setHashpreimagebase64(Buffer.from(hashPreimage).toString('base64'));
+    const claimInfoHTLCSerialized = claimInfoHTLC.serializeBinary();
+    const claimInfo = new assetLocksPb.AssetClaim();
+    claimInfo.setLockmechanism(assetLocksPb.LockMechanism.HTLC);
+    claimInfo.setClaiminfo(claimInfoHTLCSerialized);
+    return claimInfo.serializeBinary().toString();
+}
+
+// Create a SHA-256 hash over an ASCII string
+function createSHA256HashBase64(preimage)
+{
+    return crypto.createHash('sha256').update(preimage).digest('base64');
+}
+
+
+/**
+ * First/second step of a Hashed Time Lock Contract
+ * - Lock a unique asset instance using a hash
+ **/
+const createHTLC = async (
+    contract: Contract,
+    assetType: string,
+    assetID: string,
+    recipientECert: string,
+    hashValue: string,
+    expiryTimeSecs: number,
+): Promise<{ preimage: any; result: any }> => {
+
+    if (!contract)
+    {
+        logger.error("Contract handle not supplied");
+        return { preimage: "", result: false };
+    }
+    if (!assetType)
+    {
+        logger.error("Asset type not supplied");
+        return { preimage: "", result: false };
+    }
+    if (!assetID)
+    {
+        logger.error("Asset ID not supplied");
+        return { preimage: "", result: false };
+    }
+    if (!recipientECert)
+    {
+        logger.error("Recipient ECert not supplied");
+        return { preimage: "", result: false };
+    }
+
```

Comment:
Yes, I thought logs were sufficient to track errors. The Fabric gateway API works such that it either returns a chaincode execution result or throws an error if something goes wrong in the network. So that error will automatically propagate up to the app. And if you see the invocation command, we catch and throw that error already. Since these functions are executed within the application's process, throwing a log error here or in the calling function doesn't make a difference IMO.

Reply:
Not sure I quite got this, but I'll defer to your judgment.